File: src/position.py
class Position:
	def __init__(self, index, lnum, col, filename, filetext):
		self.index = index
		self.lnum = lnum
		self.col = col
		self.filename = filename
		self.filetext = filetext
	
	def advance(self, current_char=None):
		self.index += 1
		self.col += 1
		
		if current_char == "\n":
			self.lnum += 1
			self.col = 0
	
	def copy(self):
		return Position(self.index, self.lnum, self.col, self.filename, self.filetext)

File: njsgen/src/position.js
class Position {
	constructor(index, lnum, col, filename, filetext) {
		this.index = index
		this.lnum = lnum
		this.col = col
		this.filename = filename
		this.filetext = filetext
	}

	advance(current_char=null) {
		this.index += 1;
		this.col += 1;
		if (current_char == "\n") {
			this.lnum += 1
			this.col += 1
		}
	}

	copy() {
		return Position(this.index, this.lnum, this.col, this.filename, this.filetext)
	}
}